webApp/src/main/java/com/infoshare/controller/NeedRequestController.java

```diff
@@ -17,9 +28,30 @@ public NeedRequestController(NeedRequestService needRequestService) {
         this.needRequestService = needRequestService;
     }
 
+    @PostMapping("/submit-new-form")
+    public String submitNeedRequestForm(@Valid @ModelAttribute("needRequestForm") NeedRequestForm needRequestForm, BindingResult br, Model model) {
+        if (br.hasErrors()) {
+            List<TypeOfHelp> typeOfHelp = Arrays.asList(TypeOfHelp.values());
+            model.addAttribute("types", typeOfHelp);
+            return "submit-new-form";
+        } else {
+            needRequestService.createNeedRequest(needRequestForm.getName(), needRequestForm.getLocation(), needRequestForm.getPhone(), needRequestForm.getTypeOfHelp());
+            return "redirect:/need-request/print-all-need-request";
+        }
+    }
+
     @GetMapping("/create")
-    public String createNeedRequest(Model model) {
-        return getTestViewWithPageUnderConstructionMessage(model);
+    public String showNeedRequestForm(Model model) {
+      model.addAttribute(new NeedRequestForm());
+      List<TypeOfHelp> typeOfHelp = Arrays.asList(TypeOfHelp.values());
```

Comment:
Nazwa zmiennej w liczbie mnogiej.
```suggestion
      List<TypeOfHelp> typesOfHelp = Arrays.asList(TypeOfHelp.values());
```